crates/katana/primitives/src/genesis/allocation.rs

```diff
@@ -0,0 +1,282 @@
+        }
+    }
+}
+
+/// A generic non-account contract.
+#[serde_with::serde_as]
+#[derive(Debug, Default, Clone, Serialize, PartialEq, Eq)]
+pub struct GenesisContractAlloc {
+    /// The class hash of the contract.
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: ClassHash,
+    /// The amount of the fee token allocated to the contract.
+    pub balance: U256,
+    /// The initial nonce of the contract.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub nonce: Option<FieldElement>,
+    /// The initial storage values of the contract.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub storage: Option<HashMap<StorageKey, StorageValue>>,
+}
+
+/// Used mainly for development purposes where the account info including the
+/// private key is printed to the console.
+#[serde_with::serde_as]
+#[derive(Debug, Default, Clone, Serialize, PartialEq, Eq, Deref, DerefMut)]
+pub struct DevGenesisAccount {
+    /// The private key associated with the public key of the account.
+    #[serde_as(as = "UfeHex")]
+    pub private_key: FieldElement,
+    #[deref]
+    #[deref_mut]
+    #[serde(flatten)]
+    /// The inner account contract.
+    pub inner: GenesisAccount,
+}
+
+impl DevGenesisAccount {
+    /// Creates a new dev account with the given `private_key` and `class_hash`.
+    pub fn new(private_key: FieldElement, class_hash: ClassHash) -> (ContractAddress, Self) {
+        let public_key = public_key_from_private_key(private_key);
+        let (addr, inner) = GenesisAccount::new(public_key, class_hash);
+        (addr, Self { private_key, inner })
+    }
+
+    /// Creates a new dev account with the allocated `balance`.
+    pub fn new_with_balance(
+        private_key: FieldElement,
+        class_hash: ClassHash,
+        balance: U256,
+    ) -> (ContractAddress, Self) {
+        let (addr, mut account) = Self::new(private_key, class_hash);
+        account.balance = balance;
+        (addr, account)
+    }
+}
+
+/// Account contract allocated in the genesis block.
+#[serde_with::serde_as]
+#[derive(Debug, Default, Clone, Serialize, PartialEq, Eq)]
+pub struct GenesisAccount {
+    /// The public key associated with the account for validation.
+    #[serde_as(as = "UfeHex")]
+    pub public_key: FieldElement,
+    /// The class hash of the account contract.
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: ClassHash,
+    /// The amount of the fee token allocated to the account.
+    pub balance: U256,
+    /// The initial nonce of the account.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub nonce: Option<FieldElement>,
+    /// The initial storage values of the account.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub storage: Option<HashMap<StorageKey, StorageValue>>,
+}
+
+impl GenesisAccount {
+    pub fn new(public_key: FieldElement, class_hash: ClassHash) -> (ContractAddress, Self) {
+        let address = get_contract_address(
+            FieldElement::from(666u32),
```

Comment:
From where this value comes from? Do we want something special here instead?

Reply:
this is the original salt value that we've been using to generate the prefunded accounts lol

Reply:
Haha fine then let's keep it. :)